### DISCUSSION TITLE
Soft body Anchor becoming unstable - when anchoring a soft cube (not cloth) to a rigid cube

### DISCUSSION THREAD

**Discussion by aprath1** [bulletphysics/bullet3#4085] [General]
When a soft cube is anchored to a rigid cube the system is becoming unstable and flying around (using PBD mode not FEM). Anchoring cloth shown in this example [deformable_anchor.py](https://github.com/bulletphysics/bullet3/blob/6a59241074720e9df119f2f86bc01765917feb1e/examples/pybullet/examples/deformable_anchor.py) works, but when I use a soft cube it is not stable. Pleas see Code and output behavior as gif below. 
Is it somehow related to some parameter in the 'createSoftBodyAnchor' or is this a limitation of current feature?

```import pybullet as p
from time import sleep
import pybullet_data

physicsClient = p.connect(p.GUI)
p.setAdditionalSearchPath(pybullet_data.getDataPath())
# p.resetSimulation(p.RESET_USE_DEFORMABLE_WORLD)

gravZ=-10
p.setGravity(0, 0, gravZ)

planeOrn = [0,0,0,1]#p.getQuaternionFromEuler([0.3,0,0])
planeId = p.loadURDF("plane.urdf", [0,0,0],planeOrn)

boxId = p.loadURDF("cube.urdf", [0, 1, 2.5],useMaximalCoordinates = True)
# boxId = p.loadURDF("cube.urdf", basePosition = [0.5,0.45,1], globalScaling=0.25, useMaximalCoordinates = True)

ballId = p.loadSoftBody("cube.obj", basePosition = [0.37,0.37,2.1], scale = 0.25, mass = 0.5, useNeoHookean = 1, NeoHookeanMu = 400, NeoHookeanLambda = 600, NeoHookeanDamping = 0.001, useSelfCollision = 1, frictionCoeff = .5, collisionMargin = 0.001)

p.setTimeStep(0.001)
p.createSoftBodyAnchor(ballId, 2, boxId, -1, [0.45,-0.65, -0.4])
p.createSoftBodyAnchor(ballId, 5, boxId, -1, [0.5,-0.65, -0.4])

p.setPhysicsEngineParameter(sparseSdfVoxelSize=0.25)

debug = True
if debug:
  data = p.getMeshData(ballId, -1, flags=p.MESH_DATA_SIMULATION_MESH)
  print("--------------")
  print("data=",data)
  print(data[0])
  print(data[1])
  text_uid = []
  for i in range(data[0]):
      pos = data[1][i]
      uid = p.addUserDebugText(str(i), pos, textColorRGB=[1,1,1])
      text_uid.append(uid)

p.setRealTimeSimulation(1)

while p.isConnected():
  p.getCameraImage(320,200)
  
  # if debug:
  #   data = p.getMeshData(clothId, -1, flags=p.MESH_DATA_SIMULATION_MESH)
  #   for i in range(data[0]):
  #     pos = data[1][i]
  #     uid = p.addUserDebugText(str(i), pos, textColorRGB=[1,1,1], replaceItemUniqueId=text_uid[i])
  p.stepSimulation()

  p.setGravity(0,0,gravZ)
  #p.configureDebugVisualizer(p.COV_ENABLE_SINGLE_STEP_RENDERING,1)
  # sleep(1./240.)


```

![anchor_Issue](https://user-images.githubusercontent.com/88531660/144583726-819dcf75-6d4d-4462-852c-cf4a8ec7b461.gif)

**Comment by rodriguesvinay** [comment]
Hi @aprath1, I had a question. I am not able to use PBD mode (without using : p.RESET_USE_DEFORMABLE_WORLD) with .vtk file but it works well with .obj file only. What could be the reason? Thanks.

**Comment by carlostipton1319** [comment]
if you're experiencing instability issues with a soft body anchor when connecting a soft cube to a rigid cube, there could be a few factors contributing to the problem. Here are a couple of suggestions to address the [kid’s clothing  ](https://hkmart.co.nz/
)instability: Adjust Anchoring Parameters: Check the parameters of the soft body anchor and ensure they are set appropriately. Experiment with different anchor properties such as stiffness, dampening, and friction to find a balance that provides stability.
